test/Microsoft.TestPlatform.Utilities.UnitTests/InferRunSettingsHelperTests.cs

```diff
@@ -264,7 +264,75 @@ public void UpdateDesignModeOrCsiShouldModifyXmlToValueProvided(bool val)
 
             Assert.AreEqual(val.ToString(), this.GetValueOf(navigator, "/RunSettings/RunConfiguration/DesignMode"));
             Assert.AreEqual(val.ToString(), this.GetValueOf(navigator, "/RunSettings/RunConfiguration/CollectSourceInformation"));
-        }   
+        }
+
+        [DataTestMethod]
+        [DataRow("DesignMode")]
+        [DataRow("CollectSourceInformation")]
+        [DataRow("TestSessionTimeout")]
+        public void RemoveNodeFromRunConfigurationShouldNotModifyXmlIfNavigatorIsNotAtRootNode(string settingName)
+        {
+            var settings = @"<RunSettings>
+                                <RunConfiguration>
+                                    <DesignMode>False</DesignMode>
+                                    <CollectSourceInformation>False</CollectSourceInformation>
+                                    <TestSessionTimeout>1000</TestSessionTimeout>
+                                </RunConfiguration>
+                            </RunSettings>";
+            var navigator = this.GetNavigator(settings);
+            navigator.MoveToFirstChild();
+
+            switch (settingName.ToUpperInvariant())
+            {
+                case "DESIGNMODE":
+                    InferRunSettingsHelper.RemoveDesignMode(navigator);
+                    break;
+
+                case "COLLECTSOURCEINFORMATION":
+                    InferRunSettingsHelper.RemoveCollectSourceInformation(navigator);
+                    break;
+
+                case "TESTSESSIONTIMEOUT":
+                    InferRunSettingsHelper.RemoveTestSessionTimeout(navigator);
+                    break;
+            };
+
+            navigator.MoveToRoot();
+            Assert.IsTrue(navigator.InnerXml.IndexOf(settingName, StringComparison.OrdinalIgnoreCase) > 0);
+        }
+
+        [TestMethod]
+        public void RemoveDesignModeShouldRemoveDesignModeNode()
```

Comment:
No `DataTestMethod` (`RemoveNodeFromRunConfigurationShouldNotModifyXmlIfNavigatorIsNotAtRootNode`) is covering the scenario where we are sending wrong navigator, that is its a negative test. `RemoveDesignModeShouldRemoveDesignModeNode` is positive test which are checking the actual functionality of function `RemoveDesignMode`